lib/src/lib.rs

```diff
@@ -144,7 +144,18 @@
 //! [wrapping]: struct.Specification.html#structfield.wrap
 
 #![no_std]
-#![warn(unused_results, missing_docs)]
+// Opt-in for allowed-by-default lints (in alphabetical order)
+// See also: <https://doc.rust-lang.org/rustc/lints>
+#![warn(future_incompatible)]
+#![warn(let_underscore)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(rust_2018_idioms)]
+#![warn(rust_2021_compatibility)]
+#![warn(unreachable_pub)]
+#![warn(unused)]
+// Rustdoc lints
+#![warn(rustdoc::broken_intra_doc_links)]
```

Comment:
If I understand correctly, this is already warn-by-default: https://doc.rust-lang.org/rustdoc/lints.html#broken_intra_doc_links

However, the CI doesn't deny warnings in rustdoc. That's a good point. I've created #80 for that.